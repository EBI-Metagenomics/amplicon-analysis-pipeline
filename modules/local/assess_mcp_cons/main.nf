
process ASSESS_MCP_CONS {
    // Use Most Common Prefix (MCP) method to generate curves of base conservation
    tag "$meta.id"
    label 'very_light'
    // TODO: do we need this env?
    conda "${projectDir}/conf/environment.yml"

    input:
    tuple val(meta), val(fwd_flag), val(rev_flag), path(reads_merged)

    output:
    tuple val(meta), path("*mcp_cons.tsv"), optional: true, emit: mcp_cons_out

    """
    if [[ ${fwd_flag} = "auto" ]] && [[ ${rev_flag} = "auto" ]]; then
        assess_mcp_proportions -i $reads_merged -s ${meta.id}_${meta.var_region} -st FR -o ./
    elif [[ ${fwd_flag} = "auto" ]]; then
        assess_mcp_proportions -i $reads_merged -s ${meta.id}_${meta.var_region} -st F -o ./
    elif [[ ${rev_flag} = "auto" ]]; then
        assess_mcp_proportions -i $reads_merged -s ${meta.id}_${meta.var_region} -st R -o ./
    else
        touch ${meta.id}_${meta.var_region}_mcp_cons.tsv
    fi
    """

}
