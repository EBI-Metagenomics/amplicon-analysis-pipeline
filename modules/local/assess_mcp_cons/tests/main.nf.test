nextflow_process {

    name "Test Process ASSESS_MCP_CONS"
    script "../main.nf"
    process "ASSESS_MCP_CONS"
    tag "modules"
    tag "assess_mcp_cons"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    "16S-V3-V4",
                    "auto",
                    "auto",
                    file('${projectDir}/modules/local/assess_mcp_cons/tests/data/test.fastq.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.mcp_cons_out).match("mcp_cons_out") }
            )
        }

    }

}
