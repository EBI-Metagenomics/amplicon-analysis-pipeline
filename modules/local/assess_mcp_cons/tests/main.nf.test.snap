{
    "mcp_cons_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "16S-V3-V4",
                    "test_16S-V3-V4_mcp_cons.tsv:md5,875c712aadf9b21bd535e555dbb4b7e1"
                ]
            ]
        ],
        "timestamp": "2023-11-03T15:32:35.813379194"
    }
}